tests/test_spatial_analysis/test_spatial_interaction.py

```diff
@@ -482,6 +413,96 @@ def test_return_matrix_and_stratify_by_combinations(self):
                             self.assertEqual(len(result), 1)
                             self.assertIn("Full", result.keys())
 
+    def test_interaction_matrix_stratify_compute(self):
+        ax_dict = spatial_interaction(
+            self.adata,
+            "cluster_str2",
+            "Cluster Interaction Matrix",
+            stratify_by="cluster_num",
+            return_matrix=True
+            )
+
+        expected_ax_dict = {
+                    1: array([[24., 10.], [12., 2.]]),
+                    2: array([[2.,  8.], [10., 28.]])
+                }
+
+        for key, value in ax_dict[1]['Matrix'].items():
+            self.assertIn(
+                key,
+                expected_ax_dict.keys()
+            )
+
+            self.assertTrue(
+                np.array_equal(
+                    value,
+                    expected_ax_dict[key]
+                    )
+                )
+
+    def test_interaction_matrix_no_stratify_compute(self):
+        ax_dict = spatial_interaction(
+            self.adata,
+            "cluster_num",
+            "Cluster Interaction Matrix",
+            return_matrix=True
+            )
+
+        expected_array = array([[36., 24.], [12., 24.]])
+
+        self.assertTrue(

```

Comment:
@ruiheesi, would you like to assert that only one figure is returned?

Reply:
Hi George, this part of behavior is tested in the previous test. Thanks.